src/main.cc

```diff
@@ -1568,6 +1568,8 @@ SquidMain(int argc, char **argv)
 
         Mem::Init();
 
+        RunRegisteredHere(RegisteredRunner::bootstrapConfig);
```

Comment:
We should trigger this event later, as we promise in the event description -- just before parsing the configuration file. This line should be moved below the Format::Token::Init() call, just above the try/catch block that parses the configuration file.

We probably need a different bootstrapping event that happens somewhere here as well, but that configuration-unrelated event is outside this PR scope (unless you want to add two largely independent events in one PR, of course).

Reply:
I placed it here with the comment you requested removed, because all the init() calls between here and the parsing are supposed to be independent and should be runners implementing this hook.

I want to actively prohibit any of those runners accidentally working with a dependency on one of the not yet converted init() calls. Only memory allocation should be assumed working.